hyperspy/misc/utils.py

```diff
@@ -1007,8 +1010,25 @@ def stack(signal_list, axis=None, new_axis_name="stack_element", lazy=None,
         newlist = broadcast_signals(*signal_list, ignore_axis=axis_input)
 
         if axis is not None:
-            step_sizes = [s.axes_manager[axis].size for s in newlist]
-            axis = newlist[0].axes_manager[axis]
+            step_sizes = [s.axes_manager[axis_input].size for s in newlist]
+            axis = newlist[0].axes_manager[axis_input]
```

Comment:
`newlist` is an old variable that should have been renamed. Would you mind renaming it to `broadcasted_sigs`?